### PR TITLE
Deduplicate queued trashbin expire jobs

### LINKED ISSUE #20425: Jobs table is pilling up a lot of "OC\Command\CommandJob" jobs
```
SELECT class, COUNT(*) FROM oc_jobs GROUP BY class;
+-----------------------------------------------+----------+
| class                                         | COUNT(*) |
+-----------------------------------------------+----------+
| OCA\Activity\BackgroundJob\EmailNotification  |        2 |
| OCA\Activity\BackgroundJob\ExpireActivities   |        5 |
| OCA\Files_sharing\Lib\DeleteOrphanedSharesJob |        1 |
| OCA\user_ldap\lib\Jobs                        |        1 |
| OC\Command\CommandJob                         |  2198563 |
| \OCA\User_LDAP\Jobs\CleanUp                   |        1 |
+-----------------------------------------------+----------+
6 rows in set (0.47 sec)
```

Now the cron job is running endless.

Those command jobs are mostly trash bin and version expiries. I guess we need a way to parallelize those CommandJobs and trigger them separately. Otherwise big instances will have a huge problem of endless running cron.php calls and the actual cron behavior is killed.

This is on a 8.1.3 instance.

@cmonteroluque @karlitschek I rated this as a sev2-high ticket and put it in the current milestone. (feel free to change this)

@icewind1991 @PVince81 @nickvergessen @schiesbn I would really like to have your opinion on this. I think the CommandJobs can easily parallelized, because they are independent (or should be).

### PR DESCRIPTION
**Pull request by MorrisJobke**
- fixes #20425
- this removes the argument trashbin size from the expire job - it is now
  calculated in the expire job
- the queue now detects properly that the job is already queue and doesn't
  add it again

cc @rullzer @PVince81 @schiesbn 

~~I only made `getTrashbinSize` public and called it in the expire job instead of passing the size to the expire job.~~

### CODE REVIEW

File: apps/files_trashbin/command/expire.php [hunk 1]
```diff
@@ -59,7 +52,7 @@ public function handle() {
 
 		\OC_Util::tearDownFS();
 		\OC_Util::setupFS($this->user);
-		Trashbin::expire($this->trashBinSize, $this->user);
+		Trashbin::expire(Trashbin::getTrashbinSize($this->user), $this->user);
```
Comment (rullzer) [inline apps/files_trashbin/command/expire.php]: Can't you just get the trashbin size in the expire function? Then there is no need to make it public.

Comment (rullzer) [inline apps/files_trashbin/command/expire.php]: Doesn't really matter. I just like to keep things private ;)

**Comment by rullzer** [comment]
Beside my small remark this seems to do what is should now. :+1:

File: apps/files_trashbin/command/expire.php (hunk 1 above)
Comment (MorrisJobke) [inline apps/files_trashbin/command/expire.php]: Done

**Comment by MorrisJobke** [comment]
@karlitschek I will try to backport this to stable8.1 and 8.2 because in those versions that background job was introduced.

File: apps/files_trashbin/command/expire.php (hunk 1 above)
Comment (schiesbn) [inline apps/files_trashbin/command/expire.php]: I agree with @rullzer. Why not remove the trashbinSize as parameter completely and get it in the expire method.

File: apps/files_trashbin/lib/trashbin.php [hunk 2]
```diff
@@ -238,7 +235,6 @@ public static function move2trash($file_path) {
 		$trashStorage->getUpdater()->renameFromStorage($sourceStorage, $sourceInternalPath, $trashInternalPath);
 
 		if ($sizeOfAddedFiles !== false) {
```
Comment (schiesbn) [inline apps/files_trashbin/lib/trashbin.php]: We can also remove the $sizeOfAddedFiles and just make it a boolean set to false in the catch-block. Saves use one filesize() call and at least one db query.

Comment (MorrisJobke) [inline apps/files_trashbin/lib/trashbin.php]: Can we do this in a separate PR? This makes back porting easier ;)

**Comment by MorrisJobke** [comment]
stable8.2: https://github.com/owncloud/core/pull/20794

**Comment by schiesbn** [comment]
While you are on it you could also remove the size calculation in retainVersions(), it is no longer needed, so we could save some db queries here.

**Comment by schiesbn** [comment]
agreed with @MorrisJobke to do remove the unneeded size calculation in a follow-up pr to reduce the impact on the backport... so for this pr :+1:

File: apps/files_trashbin/lib/trashbin.php (hunk 2 above)
Comment (MorrisJobke) [inline apps/files_trashbin/lib/trashbin.php]: https://github.com/owncloud/core/pull/20796
